plugins/woocommerce-blocks/assets/js/base/components/combobox/index.tsx

```diff
@@ -1,62 +1,97 @@
 /**
  * External dependencies
  */
+import {
+	Combobox as AriakitCombobox,
+	ComboboxLabel,
+	ComboboxList,
+	ComboboxPopover,
+	ComboboxProvider,
+	ComboboxItem,
+	useComboboxStore,
+} from '@ariakit/react';
+import type { ComboboxProps as AriakitComboboxProps } from '@ariakit/react';
 import classnames from 'classnames';
 import { __ } from '@wordpress/i18n';
-import { useEffect, useId, useRef } from '@wordpress/element';
-import { ComboboxControl } from 'wordpress-components';
+import {
+	useCallback,
+	useEffect,
+	useId,
+	useMemo,
+	useRef,
+	useState,
+	useTransition,
+} from '@wordpress/element';
 import { ValidationInputError } from '@woocommerce/blocks-components';
-import { isObject } from '@woocommerce/types';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { VALIDATION_STORE_KEY } from '@woocommerce/block-data';
+import deprecated from '@wordpress/deprecated';
+import { useVirtualizer } from '@tanstack/react-virtual';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
+import { findExactMatchBy, findMatchingSuggestions } from './util';
 
 export interface ComboboxControlOption {
 	label: string;
 	value: string;
 }
 
-export interface ComboboxProps {
-	autoComplete?: string;
-	className?: string;
+export type ComboboxProps = Omit< AriakitComboboxProps, 'onChange' > & {
 	errorId: string | null;
 	errorMessage?: string | undefined;
-	id: string;
 	instanceId?: string;
-	label: string;
-	onChange: ( filterValue: string ) => void;
+	onChange: ( newValue: string ) => void;
 	options: ComboboxControlOption[];
-	required?: boolean;
 	value: string;
-}
+	label: string;
+};
 
 /**
- * Wrapper for the WordPress ComboboxControl which supports validation.
+ * Wrapper for the Ariakit Combobox with validation support.
  */
 const Combobox = ( {
 	id,
-	className,
 	label,
-	onChange,
 	options,
 	value,
-	required = false,
+	// Not the native onChange, a custom onChange that is called when the selected value changes.
+	onChange,
 	errorId: incomingErrorId,
-	autoComplete = 'off',
+	required = false,
+	autoComplete = 'list',
 	errorMessage = __( 'Please select a valid option', 'woocommerce' ),
+	className,
+	...restOfProps
 }: ComboboxProps ): JSX.Element => {
+	const scrollerRef = useRef( null );
 	const controlRef = useRef< HTMLDivElement >( null );
 	const fallbackId = useId();
 	const controlId = id || 'control-' + fallbackId;
 	const errorId = incomingErrorId || controlId;
 
+	let autoCompleteValue = autoComplete;
+
+	if (
+		! [ 'list', 'inline', 'both', 'none' ].includes( autoCompleteValue )
+	) {
+		deprecated( `Passing browser autocomplete hints to combobox`, {
+			since: '9.0.0',
+			plugin: '@woocommerce/components',
+			hint: 'Passing autocomplete hints to combobox has no effect, please use the values supported by Ariakit combobox: https://ariakit.org/reference/combobox#autocomplete',
+		} );
+
+		// Reset the value to the default.
+		autoCompleteValue = 'list';
+	}
```

Comment:
I'm not sure I understand this, why are moving away from the browser provided autocomplete values? Those values are meant to hint to autofill providers to what value should be in that field. Does Ariakit not provide any possible alternative? Under the hood, it seems ariakit still renders an input and that accepts autocomplete.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#autocomplete

Reply:
@senadir I find it interesting that no one doing combobox aria seems to apply these autofill hints. But I couldn't find a good reason why.

> This can be a follow up, but ideally we can get this to work. Label matching isn't enough as once you go into other locals it no longer works correctly.

Yes, I think actually, there is enough flexibility within the framework to get these working. So we could remove this warning here, and instead do a follow up to work on support for that.